File: leftwordlatest/www/my_account.js
frappe.ready(function(){
    if (frappe.session.user == 'Guest'){
        window.location.href = '/leftword_home'

    }
    $('.logout').on("click", function(e){
        e.preventDefault()
        frappe.xcall("logout").then(data=>{
            window.location.href = '/leftword_home'
        })
    })
   $('.btn-update-user').on("click", function(e) {
    $('#btnEditProfile').attr('disabled', 'true');

    frappe.xcall('leftwordlatest.web_api.user.update_user_account', {
        "first_name": $('#first-name').val(),
        "last_name": $('#last-name').val(),
        "custom_display_name": $('#display-name').val(),
        "email_id": $('#email-id').val(),
        "phone": $('#phone-number').val()
    }).then(data => {
        $('#btnEditProfile').removeAttr('disabled');
        $('#user-name').text($('#display-name').val());
        document.getElementById('phone-number').value = data["phone"] || "";
        window.location.reload();
    }).catch(err => {
        $('#btnEditProfile').removeAttr('disabled');
        console.error('Error updating user:', err);
        alert('Failed to save changes. Please try again.');
    });
});

    $('#imageUpload').on('input', function(e){
        var file = $('#imageUpload')[0].files[0];
        if (file) {
            var reader = new FileReader();
            reader.onloadend = function() {
                var base64String = reader.result;
                frappe.xcall('leftwordlatest.web_api.utils.upload_cus_img', {
                    "image": base64String,
                    "customer": $('#last-name').val(),
                }).then(data=>{
                    if(data){
                        $('.user-img').attr('src', data.file_url)
                    }
                })
                document.getElementById('base64result').textContent = base64String;
            };
            reader.readAsDataURL(file);
        } else {
            alert("Please select a file first.");
        }

    })
});

File: leftwordlatest/www/my_account.py
import frappe
from leftwordlatest.web_api.user import get_user_account

def get_context(context):
    context.no_cache = True
    load_data(context)

def load_data(context):
    if frappe.session.user != "Guest":
        user_data = get_user_account(user_details=True)
        if user_data:
            context.custom_display_name = user_data.get('custom_display_name')
            context.user_image = user_data.get('user_image')
            context.first_name = user_data.get('first_name')
            context.last_name = user_data.get('last_name')
            context.email = user_data.get('email')
            context.phone = user_data.get('phone')